screeps-game-api/src/js_collections/js_map.rs

```diff
@@ -1,13 +1,53 @@
 //! [`JsMap`]
-use std::{collections::HashMap, marker::PhantomData};
+use std::{borrow::Borrow, collections::HashMap, marker::PhantomData};
 
-use stdweb::Object;
+use stdweb::{InstanceOf, Object, Reference, ReferenceType, Value};
+
+use {
+    traits::{FromExpectedType, IntoExpectedType, TryFrom, TryInto},
+    ConversionError,
+};
 
 pub struct JsMap<K, V> {
     inner: Object,
     phantom: PhantomData<HashMap<K, V>>,
 }
 
+impl<K, V> JsMap<K, V> {
+    pub fn len(&self) -> usize {
+        self.inner.len()
+    }
+}
+
+impl<K, V> JsMap<K, V>
+where
+    Option<V>: FromExpectedType<Value>,
+{
+    /// Gets an item, panicking if the types don't match and `check-all-casts` is enabled.
+    pub fn get<Q: ?Sized>(&self, key: &Q) -> Option<V>
+    where
+        K: Borrow<Q>,
+        for<'a> &'a Q: Into<Value>,
+    {
+        js_unwrap_ref!(@{self.inner.as_ref()}[@{key.into()}])
+    }
+
+    /// Gets an item, returning an error if the types don't match and `check-all-casts` is enabled.
+    pub fn try_get<Q: ?Sized>(&self, key: &Q) -> Option<Result<V, ConversionError>>
+    where
+        K: Borrow<Q>,
+        for<'a> &'a Q: Into<Value>,
```

Comment:
This is a higher ranked trait bound: https://doc.rust-lang.org/nomicon/hrtb.html.

In english, I'd explain it as saying "for any lifetime 'a, `&'a Q` must implement `Into<Value>". In other words, `for<'a>` makes `'a` represent all lifetimes, rather than any specific one.

I'm not 100% sure why I implemented it this way, though, since in this case we can definitely just require this specific `&Q` to be `Into<Value>`. I'll change it to just use `&'a Q` and `&'a Q: Into<Value>`.

Reply:
Thanks for the insight 👍